specs/simple-serialize.md

```diff
@@ -70,21 +71,24 @@ assert value in (True, False)
 return b"\x01" if value is True else b"\x00"
 ```
 
-### Vectors, containers, lists
-
-If `value` is fixed-size:
+### Containers, vectors, lists
 
 ```python
-return "".join([serialize(element) for element in value])
-```
+# Reccursively serialize
+fixed_parts = [serialize(element) if is_fixed_size(element) else None for element in value]
+variable_parts = [serialize(element) if is_variable_size(element) else "" for element in value]
```

Comment:
Should it be empty `bytes` instead of empty `str`?
```suggestion
variable_parts = [serialize(element) if is_variable_size(element) else b"" for element in value]
```